src/NzbDrone.Core/Blacklisting/BlacklistService.cs

```diff
@@ -28,11 +30,18 @@ public BlacklistService(IBlacklistRepository blacklistRepository)
             _blacklistRepository = blacklistRepository;
         }
 
-        public bool Blacklisted(int seriesId, string sourceTitle, DateTime publishedDate)
+        public bool Blacklisted(int seriesId, ReleaseInfo release)
         {
-            var blacklisted = _blacklistRepository.Blacklisted(seriesId, sourceTitle);
+            var blacklisted = _blacklistRepository.Blacklisted(seriesId, release.Title);
+
+            if (release.DownloadProtocol == DownloadProtocol.Torrent)
+            {
+                return blacklisted.Where(b => b.Protocol == DownloadProtocol.Torrent)
+                                  .Any(b => b.Indexer.Equals(release.Indexer, StringComparison.InvariantCultureIgnoreCase));
```

Comment:
Actually, magnets contain the infohash.
But no, atm the torrentrss parser only fills in the TorrentReleaseInfo.InfoHash if explicitly provided by the feed. It doesn't atm extract the info has from magnets.